src/System.Drawing.Design.Primitives/src/System/Drawing/Design/ToolboxItem.cs

```diff
@@ -0,0 +1,1100 @@
+            }
+        }
+
+        /// <summary>
+        /// This method checks that the toolbox item is currently unlocked (read-write) and
+        /// throws an appropriate exception if it isn't.
+        /// </summary>
+        protected void CheckUnlocked()
+        {
+            if (Locked)
+                throw new InvalidOperationException(SR.GetString(SR.ToolboxItemLocked));
+        }
+
+        /// <summary>
+        /// Creates objects from the type contained in this toolbox item.
+        /// </summary>
+        /// <returns></returns>
+        public IComponent[] CreateComponents()
+        {
+            return CreateComponents(null);
+        }
+
+        /// <summary>
+        /// Creates objects from the type contained in this toolbox item.  If designerHost is non-null
+        /// this will also add them to the designer.
+        /// </summary>
+        public IComponent[] CreateComponents(IDesignerHost host)
+        {
+            OnComponentsCreating(new ToolboxComponentsCreatingEventArgs(host));
+            IComponent[] comps = CreateComponentsCore(host, new Hashtable());
+            if (comps != null && comps.Length > 0)
+            {
+                OnComponentsCreated(new ToolboxComponentsCreatedEventArgs(comps));
+            }
+            return comps;
+        }
+
+        /// <summary>
+        /// Creates objects from the type contained in this toolbox item.  If designerHost is non-null
+        /// this will also add them to the designer.
+        /// </summary>
+        /// <returns></returns>
+        public IComponent[] CreateComponents(IDesignerHost host, IDictionary defaultValues)
+        {
+            OnComponentsCreating(new ToolboxComponentsCreatingEventArgs(host));
+            IComponent[] comps = CreateComponentsCore(host, defaultValues);
+            if (comps != null && comps.Length > 0)
+            {
+                OnComponentsCreated(new ToolboxComponentsCreatedEventArgs(comps));
+            }
+            return comps;
+        }
+
+        /// <summary>
+        /// Creates objects from the type contained in this toolbox item.  If designerHost is non-null
+        /// this will also add them to the designer.
+        /// </summary>
+        protected virtual IComponent[] CreateComponentsCore(IDesignerHost host)
+        {
+            ArrayList comps = new ArrayList();
+
+            Type createType = GetType(host, AssemblyName, TypeName, true);
+            if (createType != null)
+            {
+                if (host != null)
+                {
+                    comps.Add(host.CreateComponent(createType));
+                }
+                else if (typeof(IComponent).IsAssignableFrom(createType))
+                {
+                    comps.Add(TypeDescriptor.CreateInstance(null, createType, null, null));
+                }
+            }
+
+            IComponent[] temp = new IComponent[comps.Count];
+            comps.CopyTo(temp, 0);
+            return temp;
```

Comment:
Nit: again, we probably want to leave code as-is for the port, but in the future we could use `List<T>`, or if not for some reason, here use `return comps.ToArray(typeof(IComponent));` just to simplify.

Reply:
I am keeping this snippet as is for now , for this port.  There are more than one instances in this pattern in this commit.